content/blog/2022/headed-playwright-in-wsl-docker.md

```diff
@@ -70,39 +70,52 @@ You most likely already have WSL and WSLg installed if you are running [Docker
 Desktop](https://www.docker.com/products/docker-desktop/) in recent builds of
 Windows 10 and 11.
 
-Let's start by verifying that WSL and WSLg are installed and running:
-
-```powershell
-wsl.exe --version
-
-WSL version: 1.0.0.0
-Kernel version: 5.15.74.2
-WSLg version: 1.0.47
-MSRDC version: 1.2.3575
-Direct3D version: 1.606.4
-DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
-Windows version: 10.0.22621.819
+Let's start by verifying that WSL and WSLg are installed and running. First,
+launch "WSL" from you Start Menu. A Linux terminal window should open (most
+likely a recent version of Ubuntu). Then verify that the directory `/mnt/wslg/`
+exists and contains these files inside the Linux filesystem:
+
+```bash
+ls -a -w 1 /mnt/wslg
+
+.
+..
+.X11-unix
+PulseAudioRDPSink
+PulseAudioRDPSource
+PulseServer
+distro
+doc
+dumps
+pulseaudio.log
+runtime-dir
+stderr.log
+versions.txt
+weston.log
+wlog.log
 ```
 
-If the command returns without errors and you see a line with `WSLg version:
-X.X.X`, then you're good to go! According to [this
-guide](https://github.com/microsoft/wslg/blob/main/samples/container/Containers.md),
-we only need to do two things:
+If you don't see "WSL" in your Start menu, or the `ls` command above fails with
+`No such file or directory`, then your system is missing WSL entirely or is
+running an old version. Visit the [Microsoft Store](https://aka.ms/wslstorepage)
```

Comment:
Not surprisingly, this link only works on Windows? Might be worth noting that.

Related, should the headline of this post be more clear that it's Windows-specific? And/or add a new "Windows" tag?

Reply:
I would love to provide instructions for other operating systems, but Windows is all I have access to. Would you be willing to give something like [this](http://mamykin.com/posts/running-x-apps-on-mac-with-docker/) a try to include it?

In the meantime, I added a quick note at the top on ca199295f528b291dced2460ecc19cb604b93905. Feel free to switch to something in the title or a tag.

Reply:
> I would love to provide instructions for other operating systems, but Windows is all I have access to. Would you be willing to give something like [this](http://mamykin.com/posts/running-x-apps-on-mac-with-docker/) a try to include it?

It worked! I needed to:

1. Install XQuartz: `brew install --cask xquartz`
2. Open XQuartz, Preferences -> Security, check "Allow connections from network clients"
3. Restart my computer (restarting XQuartz didn't seem to work)
4. Start XQuartz with `xhost +localhost`
5. Edit Docker settings to give access to `/tmp/.X11-unix`: Preferences -> Resources -> File sharing
6. Use `host.docker.internal:0` instead of just `:0`

```bash
docker run --rm \
-e DISPLAY=host.docker.internal:0 \
-v /tmp/.X11-unix:/tmp/.X11-unix \
mcr.microsoft.com/playwright:v1.28.0 npx -y playwright open google.com
```

Reply:
Awesome, I incorporated that in 3863ea2871552ae1848294f65cee2fdea5c5c308